src/components/ProblemSetBlock.tsx

```diff
@@ -171,6 +173,7 @@ export const ProblemSetBlock = ({ waitForEvent, fireSuccessEvent, fireLearningOp
       contentId: prob.contentId,
       solution: prob.solution,
       retryLimit: prob.retryLimit,
```

Comment:
It does work - I originally opted for being a little bit more explicit in lieu of DRYing things up. It may not be obvious if you only look at the problem types, but if you look for how they get their props I think it would not be hard to understand so long as you understand the logic behind `??`.

I've updated to match the pattern noted above.